packages/docusaurus/lib/server/index.js

```diff
@@ -55,35 +55,39 @@ module.exports = async function load(siteDir, cliOptions = {}) {
   // These can be overriden in plugins/ through component swizzling.
   // However, we alias it here first as a fallback.
   const themeFallback = path.resolve(__dirname, '../client/theme-fallback');
-  let themeAliases = await loadTheme(themeFallback);
+  const fallbackAliases = await loadTheme(themeFallback);
 
-  // create theme alias from plugins
-  await Promise.all(
+  // Create theme alias from plugins.
+  const pluginThemeAliases = await Promise.all(
```

Comment:
Changing it such that we only combine the aliases after all the promises are resolved. This way we still resolve in parallel but we preserve the order.